SSA/Projects/InstCombine/HackersDelight.lean

```diff
@@ -105,6 +137,26 @@ theorem and_eq_not_or_sub_not {x y : BitVec w} :
     x &&& y = (~~~ x ||| y) - ~~~ x := by
   sorry
 
+theorem xor_le_or {x y : BitVec w} :
+    x ^^^ y ≤ x ||| y := by
+  sorry
+
+theorem and_le_not_xor {x y : BitVec w} :
+    x &&& y ≤ ~~~(x ^^^ y) := by
+  sorry
+
+theorem or_le_add {x y : BitVec w} (h : x.toFin + y.toFin < 2^w) :
```

Comment:
Semantically, it's the same, but it may be slightly clearer to give a label to the property of "no overflows"

```suggestion
def AdditionNoOverflows? (x y : BitVec w) : Prop := (x.adc y false).1

theorem or_le_add  (h : AdditionNoOverflows? x y) :
    x ||| y ≤ x + y := by
  sorry
```